aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/customize/s3/S3Decorator.kt

```diff
@@ -23,17 +28,24 @@ import software.amazon.smithy.rust.codegen.smithy.customize.RustCodegenDecorator
 import software.amazon.smithy.rust.codegen.smithy.generators.LibRsCustomization
 import software.amazon.smithy.rust.codegen.smithy.generators.LibRsSection
 import software.amazon.smithy.rust.codegen.smithy.letIf
+import software.amazon.smithy.rust.codegen.smithy.protocols.AllowInvalidXmlRoot
 import software.amazon.smithy.rust.codegen.smithy.protocols.ProtocolMap
 import software.amazon.smithy.rust.codegen.smithy.protocols.RestXml
 import software.amazon.smithy.rust.codegen.smithy.protocols.RestXmlFactory
 import software.amazon.smithy.rustsdk.AwsRuntimeType
+import java.util.logging.Logger
 
 /**
  * Top level decorator for S3
  */
 class S3Decorator : RustCodegenDecorator<ClientCodegenContext> {
-    override val name: String = "S3ExtendedError"
+    override val name: String = "S3"
     override val order: Byte = 0
+    private val logger: Logger = Logger.getLogger(javaClass.name)
+    private val invalidXmlRootAllowList = setOf(
+        // API returns GetObjectAttributes_Response_ instead of Output
+        ShapeId.from("com.amazonaws.s3#GetObjectAttributesOutput"),
```

Comment:
I took a brief stroll through botocore to see if there were any other obvious operations that have the wrong shape, and we may want to check `s3#GetBucketLocation`, `s3#ListObjects`, `s3#ListObjectVersions`, and `ec2#GetConsoleOutput` (this last one is using the ec2Query protocol, which still has XML responses). Not saying these ones meet the criteria, but that I see [special code in botocore parsing these XML responses in botocore](https://github.com/boto/botocore/blob/develop/botocore/handlers.py). If it's easy to check these, then we should, but probably not worth spending a ton of time on.

Reply:
I've broken this work out into a separate issue: #1668